halo2_frontend/src/dev/cost_model.rs

```diff
@@ -279,13 +280,35 @@ pub fn from_circuit_to_model_circuit<
     options.into_model_circuit::<COMM, SCALAR>(comm_scheme)
 }
 
-/// Given a Plonk circuit, this function returns [CostOptions]
+fn run_mock_prover_with_fallback<F: Ord + Field + FromUniformBytes<64>, C: Circuit<F>>(
+    circuit: &C,
+    instances: Vec<Vec<F>>,
```

Comment:
Take a slice ref instead of a vec.

Reply:
Not saying that this is the right way of doing it, but its used all over the library (representing instances as Vec<Vec<>>). I'll leave it for now and open an issue to handle it later.